### PR TITLE
Policy for using GitHub Apps, Extensions, and bots

### PR DESCRIPTION
**Pull request by SergeyKanzhelev**

### CODE REVIEW

File: docs/using-github-extensions.md [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+In OpenTelemetry it is critically important to preserve code integrity, including copyright.
+
+No Extensions, Apps, or bots that can arbitrarily modify code are allowed. Read-only code access, opening pull requests from another fork, or bug triage bots can be considered to be installed.
+
+## Requesting installing of Apps, Extensions, or bots
+
+- The default answer for installing third party tools in the org is "NO". Justification needs to be provided and supported by multiple org members.
+- Open an issue at https://github.com/open-telemetry/community/issues
+  - Include reasoning and SIG/SIGs requesting this
```
Comment (yurishkuro) [inline docs/using-github-extensions.md:11]:
```suggestion
  - Include reasoning and SIG(s) requesting this
```

File: docs/using-github-extensions.md [hunk 2]
```diff
@@ -0,0 +1,30 @@
+# Using GitHub Apps, Extensions, and bots
+
+In OpenTelemetry it is critically important to preserve code integrity, including copyright.
+
+No Extensions, Apps, or bots that can arbitrarily modify code are allowed. Read-only code access, opening pull requests from another fork, or bug triage bots can be considered to be installed.
```
Comment (yurishkuro) [inline docs/using-github-extensions.md:5]:
```suggestion
Extensions, apps, or bots that can **arbitrarily modify code are NOT allowed**. Read-only code access, opening pull requests from another fork, or bug triage bots can be considered to be installed.
```

File: docs/using-github-extensions.md [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+## Requesting installing of Apps, Extensions, or bots
+
+- The default answer for installing third party tools in the org is "NO". Justification needs to be provided and supported by multiple org members.
+- Open an issue at https://github.com/open-telemetry/community/issues
+  - Include reasoning and SIG/SIGs requesting this
+  - Requests from maintainers are typically carry higher weight, please make sure to discuss in the SIG you participate.
```
Comment (yurishkuro) [inline docs/using-github-extensions.md:12]: ****
```suggestion
  - Requests from maintainers typically carry higher weight; please make sure to discuss in the SIG you participate.
```

File: docs/using-github-extensions.md [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+  - List permissions this extension, app, or bot requests.
```
Comment (yurishkuro) [inline docs/using-github-extensions.md:13]:
```suggestion
  - List permissions required by this extension, app, or bot.
```

File: docs/using-github-extensions.md [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+  - If possible: point to other uses of this extensions in OSS.
+- GC member needs to approve with no other GC members raising concerns. It is recommended to discuss each extension, app, or bot that is about to be installed at GC meeting for awareness.
+- Once GC approval received, TC member will install the extension.
+
+## Writing your GitHub Actions pipelines
+
+Many GitHub Action workflows does not require a dedicated GutHub Account created. Good example is an PR and issues auto-assign workflows like [one used in specifications repo](https://github.com/open-telemetry/opentelemetry-specification/blob/main/.github/workflows/auto-assign-issue.yml).
```
Comment (yurishkuro) [inline docs/using-github-extensions.md:20]:
```suggestion
Many GitHub Action workflows do not require a dedicated GitHub account. Good examples are auto-assign workflows for issues and PRs such as [the one used in specifications repo](https://github.com/open-telemetry/opentelemetry-specification/blob/main/.github/workflows/auto-assign-issue.yml).
```

File: docs/using-github-extensions.md [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+
+There are cases when a dedicated account is needed to perform some higher privilege operations. In this cases the recommendation for maintainers  to use their accounts. See [example](https://github.com/open-telemetry/opentelemetry-specification/blob/main/.github/workflows/publish-schemas.yml).
```
Comment (yurishkuro) [inline docs/using-github-extensions.md:22]:
```suggestion
There are cases when a dedicated account is needed to perform some higher privilege operations. In this cases the recommendation for maintainers is to use their accounts. See [example](https://github.com/open-telemetry/opentelemetry-specification/blob/main/.github/workflows/publish-schemas.yml).
```

File: docs/using-github-extensions.md [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+
+Bots accounts are not recommended, as they cannot sign the CLA. If bot account needs to be used, it should be clear who owns this account and who signed CLA on behalf of this bot.
```
Comment (yurishkuro) [inline docs/using-github-extensions.md:24]:
```suggestion
Bot accounts are not recommended, as they cannot sign the CLA. If bot account needs to be used, it should be clear who owns this account and who signed CLA on behalf of this bot.
```

File: docs/using-github-extensions.md [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+- GC member needs to approve with no other GC members raising concerns. It is recommended to discuss each extension, app, or bot that is about to be installed at GC meeting for awareness.
+- Once GC approval received, TC member will install the extension.
+
+## Writing your GitHub Actions pipelines
+
+Many GitHub Action workflows do not require a dedicated GitHub account. Good examples are auto-assign workflows for issues and PRs such as [the one used in specifications repo](https://github.com/open-telemetry/opentelemetry-specification/blob/main/.github/workflows/auto-assign-issue.yml).
+
+There are cases when a dedicated account is needed to perform some higher privilege operations. In this cases the recommendation for maintainers is to use their accounts. See [example](https://github.com/open-telemetry/opentelemetry-specification/blob/main/.github/workflows/publish-schemas.yml).
+
+Bot accounts are not recommended, as they cannot sign the CLA. If bot account needs to be used, it should be clear who owns this account and who signed CLA on behalf of this bot.
```
Comment (trask) [inline docs/using-github-extensions.md:24]: there was recent ask in maintainer meeting related to using bot accounts for release automation, and the GC meeting notes from 2021-09-02 mention:

> CNCF easycla team gave permission to exempt specific branches or branch prefixes from the pull request requirement to create/edit branches given that no code on the branch introduces new copywritable content (release automation)

is this something that makes sense to address here?

Comment (SergeyKanzhelev) [reply docs/using-github-extensions.md:24]: @dyladan maybe you can add information about this as a follow up PR?

**Review by yurishkuro — APPROVED** [review]

File: docs/using-github-extensions.md (hunk 8 above)
Comment (lizthegrey) [reply docs/using-github-extensions.md:24]: I'm the person who's been our liaison to the easycla team, I can explain.

```
Bot accounts are not recommended, as they require special work to pass CLA checks. If pushing automatically-generated, non-copyrightable code using a bot account is required, an explanation should be sent to the Governance Committee for review and forwarding to the EasyCLA team to exempt the bot's commits from the CLA requirement.
```

Comment (SergeyKanzhelev) [reply docs/using-github-extensions.md:24]:
```suggestion
Bot accounts are not recommended, as they require special work to pass CLA checks. If pushing automatically-generated, non-copyrightable code using a bot account is required, an explanation should be sent to the Governance Committee for review and forwarding to the EasyCLA team to exempt the bot's commits from the CLA requirement.
```

**Pull request merged**